#!/usr/bin/env python

PACKAGE = "navtech_projection"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# gen.add("radar_distance_resolution",    double_t,   0, "distance resolution", 50,  0, 100)
gen.add("interpolate",             bool_t,     0, "interpolate pixels, smoothing the edge between polar angles",       False)
gen.add("enable_cartesian",             bool_t,     0, "publish cartesian image",               True)
gen.add("cart_image_size",              int_t,      0, "image size of output cartesian image",  5712,   0,      5712)
gen.add("cart_range_x_min",               double_t,   0, "max range of output cartesian image",   -400.,   -400.,      400.)
gen.add("cart_range_x_max",               double_t,   0, "max range of output cartesian image",   400.,   -400.,      400.)
gen.add("cart_range_y_min",               double_t,   0, "max range of output cartesian image",   -400.,   -400.,      400.)
gen.add("cart_range_y_max",               double_t,   0, "max range of output cartesian image",   400.,   -400.,      400.)
gen.add("enable_pointcloud",            bool_t,     0, "publish point cloud",                   True)
gen.add("pc_range_x_min",                 double_t,   0, "max range of rendered point cloud",     -400.,   -456.,     456.)
gen.add("pc_range_x_max",                 double_t,   0, "max range of rendered point cloud",     400.,   -456.,     456.)
gen.add("pc_range_y_min",                 double_t,   0, "max range of rendered point cloud",     -400.,   -456.,     456.)
gen.add("pc_range_y_max",                 double_t,   0, "max range of rendered point cloud",     400.,   -456.,     456.)
gen.add("pc_pixel_resolution",          double_t,   0, "point cloud resolution",                0.175,    0.1,    3.2)
gen.add("pc_intensity_threshold",       int_t,      0, "threshold of point cloud intensity",    0,      0,      255)

exit(gen.generate(PACKAGE, "radar_image_projection", "params"))